src/__tests__/Graph-test.js

```diff
@@ -0,0 +1,207 @@
+import Graph from '../Graph'
+import assert from 'assert'
+
+describe('Graph', function() {
+
+  it ('adjusts the focal point when adding a node', function() {
+    let graph = new Graph()
+
+    graph.add('one')
+    assert.equal(graph.focus, 'one')
+
+    graph.add('two')
+    assert.equal(graph.focus, 'two')
+  })
+
+  it ('does not add undefined values', function() {
+    let graph = new Graph()
+
+    assert.throws(() => graph.add(undefined), /Graph unable to add node/)
+  })
+
+  it ('does not add a node twice', function() {
+    let graph = new Graph()
+
+    graph.add('one')
+    graph.add('one')
+
+    assert.equal(graph.nodes.length, 1)
+  })
+
+  it ('adjusts the focal point back when adding an existing node', function() {
+    let graph = new Graph()
+
+    graph.add('one')
+    graph.add('two')
+    graph.add('one')
+
+    assert.equal(graph.focus, 'one')
+  })
+
+  it ('can remove all traces of a node', function() {
+    let graph = new Graph()
+
+    graph.connect('one', 'two')
+    graph.connect('three', 'four')
+
+    graph.remove('three')
+
+    assert(graph.nodes.indexOf('three') < 0)
+
+    graph.edges.forEach(function(edge) {
+      assert(edge.indexOf('three') < 0)
+    })
+  })
+
+  it ('removing a focused node moves focus to the next position', function() {
+    let graph = new Graph()
+
+    graph.connect('one', 'two')
+    graph.remove('one')
+
+    assert.equal(graph.focus, 'two')
+  })
+
+  it ('removing a focused node sets focus to false if there is no next node', function() {
+    let graph = new Graph()
+
+    graph.add('one')
+    graph.remove('one')
+
+    assert.equal(graph.focus, false)
```

Comment:
:+1:. That is a really good idea... `focus`, `after`, and `before` should be set or return `null`, not `false`